s/nub-context/nub-context.ts

```diff
@@ -0,0 +1,70 @@
+import {dispatchCustomEvent} from "../nub-keyboard/utils/dispatchCustomEvent.js"
+import {keys} from "../nub-keyboard/utils/keys.js"
+import {styles} from "./style.css.js"
+
+export class NubContext extends HTMLElement {
+	shadow = this.attachShadow({
+		mode: "open",
+		delegatesFocus: false,
+	})
+
+	constructor() {
+		super()
+		let editable = false
+		this.shadow.innerHTML = `
+		<div>
+		<button class="flip">flip mode</button>
+		<slot>
+		</slot>
+		</div>
+		`
+		const style = document.createElement("style")
+		style.textContent = styles
+		this.shadowRoot?.append(style)
+		const flipButton = this.shadowRoot?.querySelector<HTMLElement>(".flip")
+		const nubContext = document.querySelector("nub-context")
+		nubContext?.addEventListener("nub-input", (e: any) => {
+			const detail = e.detail
+			if (detail?.actionName) {
+			dispatchCustomEvent(this, detail.actionName)
+			}
+		})
+		nubContext?.addEventListener("nub-action", (e: any) => {
+			console.log(e.detail)
+		})
+		// this.prepare({
+		// 	storage: window.localStorage,
+		// 	actions: {
+		// 		forward: "Button"
+		// 	},
+		// 	defaultBinds: {
+		// 		forward: ["W", "e"]
+		// 	}
+		// })
+		flipButton?.addEventListener("pointerdown", () => {
+			if (this.hasAttribute('edit-mode')) {
+				this.removeAttribute("edit-mode")
+			} else {
+				this.setAttribute("edit-mode", "")
+			}
+		})
+	}
+	prepare({storage, actions, defaultBinds}: {
+		storage: Storage,
+		actions: {
+			[key: string]: string
+		},
+		defaultBinds: {
+			[key: string]: string[]
+		}
+	}) {
+		console.log(actions, defaultBinds)
+		for (const [actionName, defaultKey] of Object.entries(defaultBinds)) {
+			defaultKey.map(key => {
+				if (keys[key.toUpperCase()]) {
+					keys[key.toUpperCase()].actionName = actionName
```

Comment:
:warning: on this line, we actually mutating a module-level object that we imported

it's a golden rule of programming -- never mutate anything at global or module level

so we'll have to find a better way to communicate this information about the action name

perhaps we need to rethink what it means to "bind keys"